### PR TITLE
Drop json gem support

### LINKED ISSUE #112: Support for Ruby 2.4
Re:

``` 
s.add_dependency 'json', '~> 1.3' 
```

json on 2.4 seems to be at 2.0.2, trying to install an old json like 1.8.3 on 2.4 will fail. 

is there any reason for this explicit dependency?

### PR DESCRIPTION
**Pull request by arunagw**
Fixes #112 #108

### CODE REVIEW

**Comment by SamSaffron** [comment]
awesome thanks!

**Comment by unasuke** [comment]
@arunagw 
Great commits.
But, tests on travis ci are failed because run tests in ruby 1.9.3 .

I suggest drop support in eol ruby versions(1.9.x and 2.0.x).

File: omniauth-twitter.gemspec [hunk 1]
```diff
@@ -16,8 +16,8 @@ Gem::Specification.new do |s|
   s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
-
-  s.add_dependency 'json', '~> 1.3'
+  s.required_ruby_version = Gem::Requirement.new('>= 1.9.3')
   s.add_dependency 'omniauth-oauth', '~> 1.1'
+  s.add_dependency 'rack', '~> 1.6'
```
Comment (knu) [inline omniauth-twitter.gemspec]: This version restriction for rack conflicts with that of Rails 5 (~> 2.0).  Is that intentional?

**Comment by arunagw** [comment]
#114 merged

**Pull request closed without merging**